### PR TITLE
Fix #2950: prefer inner error

### LINKED ISSUE #2950: Error reporting issue on extracting inner class (Applications.scala)
```scala
class T1 {
  case class Foo(x: Int, xs1: List[String], xs2: List[String])
}

object T2 {
  val foo: T1#Foo = ???

  val Foo(x1, xs1, xs2) = foo
}
```

While not a valid usage of extractors, this reports some weird errors:

```scala
-- Error: local/ext.scala:8:6 --------------------------------------------------
8 |  val Foo(x1, xs1, xs2) = foo
  |      ^^^
  |      Foo cannot be used as an extractor in a pattern because it lacks an
  |      unapply or unapplySeq method

-- [E044] Syntax Error: local/ext.scala:8:2 ------------------------------------
8 |  val Foo(x1, xs1, xs2) = foo
  |  ^
  |  recursive value x1 needs type

longer explanation available when compiling with `-explain`
-- [E045] Syntax Error: local/ext.scala:8:23 -----------------------------------
8 |  val Foo(x1, xs1, xs2) = foo
  |                       ^
  |                       cyclic reference involving value $1$

longer explanation available when compiling with `-explain`
```

The extraction could've been done like:

```scala
val t1 = new T1
val t1.Foo(x1, xs1, xs2) = foo
```

### PR DESCRIPTION
**Pull request by liufengyun**
Fix #2950: prefer inner error

The inner error is supposed to be more informative thus should be preferred.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/typer/Applications.scala [hunk 1]
```diff
@@ -941,7 +941,7 @@ trait Applications extends Compatibility { self: Typer with Dynamic =>
     val Apply(qual, args) = tree
 
     def notAnExtractor(tree: Tree) =
-      if (tree.tpe.isErroneous) tree
+      if (!tree.tpe.isError && tree.tpe.isErroneous) tree
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/typer/Applications.scala]: Why the special case for `isError` here ?

Comment (liufengyun) [reply compiler/src/dotty/tools/dotc/typer/Applications.scala]: `isError = true` implies an error on the outer-most tree, thus `NotAnExtractor` error should be preferred.

Comment (smarter) [reply compiler/src/dotty/tools/dotc/typer/Applications.scala]: > thus NotAnExtractor error should be preferred

Did you mean "should _not_ be preferred" ? You're not using the error from NotAnExtractor when isError is true.

Comment (liufengyun) [reply compiler/src/dotty/tools/dotc/typer/Applications.scala]: The check `!tree.tpe.isError && tree.tpe.isErroneous ` ensures that `isError = true` will go to the 2nd branch, which is a `NotAnExtractor` error.

Comment (smarter) [reply compiler/src/dotty/tools/dotc/typer/Applications.scala]: Ah right, got confused, but can you add a comment that gives an example where NotAnExtractor should be preferred to get a good error message ?

**Pull request merged**